File: NumberOfDiscIntersections/NumberOfDiscIntersections.py
def solution(A):
    size = len(A)
    result = 0
    for i in range(size):
        a = A[i]
        if i + a >= size - 1:
            result += size - 1 - i
        else:
            result += a
            fromJ = i + a + 1
            for j in range(fromJ, size):
                if A[j] > j - fromJ:
                    result += 1
        if result > 10_000_000:
            return -1
    return result

File: NumberOfDiscIntersections/NumberOfDiscIntersections.java
class Solution {
    public int solution(int[] A) {
        int size = A.length;
        int result = 0;
        for (int i = 0; i < size; i++) {
            int a = A[i];
            if (i + a >= size - 1) {
                result += size - 1 - i;
            } else {
                result += a;
                int fromJ = i + a + 1;
                for (int j = fromJ; j < size; j++) {
                    if (A[j] > j - fromJ) {
                        result++;
                    }
                }
            }
            if (result > 10_000_000) {
                return -1;
            }
        }
        return result;
    }
}